modules/core/src/test/java/com/google/refine/model/changes/ColumnRenameChangeTest.java

```diff
@@ -0,0 +1,218 @@
+
+package com.google.refine.model.changes;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.io.Serializable;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Properties;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import com.google.refine.ProjectManager;
+import com.google.refine.RefineTest;
+import com.google.refine.model.Column;
+import com.google.refine.model.Project;
+import com.google.refine.util.Pool;
+
+public class ColumnRenameChangeTest extends RefineTest {
+
+    private Project project;
+
+    @BeforeTest
+    public void init() {
+        logger = org.slf4j.LoggerFactory.getLogger(this.getClass());
+    }
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        project = createProject(
+                new String[] { "original_column", "other_column" },
+                new Serializable[][] {
+                        { "data1", "value1" },
+                        { "data2", "value2" }
+                });
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        if (project != null) {
+            ProjectManager.singleton.deleteProject(project.id);
+        }
+    }
+
+    @Test
+    public void testSimpleRename() throws Exception {
+        ColumnRenameChange change = new ColumnRenameChange("original_column", "new_column");
+        change.apply(project);
+
+        Column renamed = project.columnModel.getColumnByName("new_column");
+        assertNotNull(renamed);
+        assertEquals(renamed.getName(), "new_column");
+    }
+
+    @Test
+    public void testRevertRename() throws Exception {
+        ColumnRenameChange change = new ColumnRenameChange("original_column", "new_column");
+        change.apply(project);
+        change.revert(project);
+
+        Column original = project.columnModel.getColumnByName("original_column");
+        assertNotNull(original);
+        assertEquals(original.getName(), "original_column");
+    }
+
+    @Test
+    public void testEscapeNewlinesRoundTrip() throws Exception {
+        String oldName = "col\nwith\nlines";
+        String newName = "new\ncol";
+
+        ColumnRenameChange change = new ColumnRenameChange(oldName, newName);
+
+        Writer writer = new StringWriter();
+        change.save(writer, new Properties());
```

Comment:
```suggestion
        History.writeOneChange(writer, change, new Pool());
```

Using the same method as the production code will make this more authentic